defi/src/storeTvlInterval/computeTVL.ts

```diff
@@ -1,98 +1,110 @@
-      }
-    })
-    .filter((item) => item !== undefined) as string[];
+
+  return balances;
+}
+
+async function fetchTokenData(
+  timestamp: "now" | number,
+  balances: { [address: string]: string }
+) {
   const readRequests = [];
-  for (let i = 0; i < readKeys.length; i += 100) {
-    const body = {
-      "coins": readKeys.slice(i, i + 100),
-    } as any
-    if(timestamp !== "now"){
-      body.timestamp = timestamp;
-    }
+  const burl = "https://coins.llama.fi/prices/";
+  const historical =
+    timestamp == "now" ? "current/" : `historical/${timestamp}/`;
+
+  for (let i = 0; i < Object.keys(balances).length; i += step) {
+    const coins = Object.keys(balances)
+      .slice(i, i + step)
+      .reduce((p, c) => p + `${c},`, "");
     readRequests.push(
-      fetch("https://coins.llama.fi/prices", {
-        method: "POST",
-        body: JSON.stringify(body)
-      }).then((r) => r.json()).then(r=>{
-        return Object.entries(r.coins).map(
-        ([PK, value])=>({
-          ...(value as any),
-          PK
-        })
+      fetch(`${burl}${historical}${coins}`, { method: "GET" }).then((r) =>
+        r.json()
       )
-    })
     );
   }
-  const tokenData = ([] as any[]).concat(...(await Promise.all(readRequests)));
+
+  const responses: any[] = await Promise.all(readRequests);
+  return responses.map((g) => g.coins);
+}
+
+const confidenceThreshold = 0.5;
+const step = 40;
+export default async function (
+  balances: { [address: string]: string },
+  timestamp: "now" | number,
+  staleCoins: any
+) {
+  staleCoins; // can get rid of stalecoins type?
+  balances = normalizeBalances(balances);
+
+  const tokenData = await fetchTokenData(timestamp, balances);
+
   let usdTvl = 0;
   const tokenBalances = {} as Balances;
   const usdTokenBalances = {} as Balances;
-  const now = timestamp === "now" ? Math.round(Date.now() / 1000) : timestamp;
+
   tokenData.forEach((response) => {
-    if (Math.abs(response.timestamp - now) > 3600*1.2) { // 1.2 hours
-      addStaleCoin(staleCoins, response.PK, response.symbol, response.timestamp);
-    }
-    if (Math.abs(response.timestamp - now) < DAY) {
-      PKsToTokens[response.PK].forEach((address) => {
-        const balance = balances[address];
-        const { price, decimals } = response;
-        let symbol:string, amount:number;
-        if (response.PK.startsWith('coingecko:')) {
-          symbol = address;
-          amount = Number(balance);
-        } else {
-          symbol = (response.symbol as string).toUpperCase();
-          amount = new BigNumber(balance).div(10 ** decimals).toNumber();
-        }
-        const usdAmount = amount * price;
-        tokenBalances[symbol] = (tokenBalances[symbol] ?? 0) + amount;
-        usdTokenBalances[symbol] = (usdTokenBalances[symbol] ?? 0) + usdAmount;
-        usdTvl += usdAmount;
-      });
-    } else {
-      console.error(`Data for ${response.PK} is stale`);
-    }
+    Object.keys(response).forEach((address) => {
```

Comment:
i assume the point of removing stalecoins tracking is that now we already remove stale coins on coins backend but the big issue is that we dont track them there + cant track tvl impact there

Reply:
in my latest PR i use stalecoins to track how much TVL is associated with each stale coin: https://github.com/DefiLlama/defillama-server/pull/1101/files#diff-2ca08735ce63a32600edb7f03c852820e7433320a15f096e4264cb96eee5a717L70

Reply:
can probably figure out some other way since .timestamp is also provided by the new api, so not a deal breaker for this PR

Reply:
had a go at reimplementing this - I guess the new endpoint wont return symbols and decimals for stale coins so we'd need to requery the DB, or just use the PK as reference

Reply:
requerying the db is not a problem imo, where can i find your re-implementation?